gcloud/datastore/__init__.py

```diff
@@ -69,6 +69,50 @@
 _DATASET_ENV_VAR_NAME = 'GCLOUD_DATASET_ID'
 
 
+def _find_true_dataset_id(dataset_id, connection=None):
```

Comment:
We likely need to make this method public, to allow applications which don't use the inferred dataset ID to normalize their project ID -> dataset ID.